djangocms_blog/search_indexes.py

```diff
@@ -47,32 +47,32 @@ def get_model(self):
         return Post
 
     def get_search_data(self, post, language, request):
-        optional_attributes = []
-        abstract = post.safe_translation_getter('abstract')
+        description = post.get_description()
+        keywords = post.get_keywords()
+
         text_bits = [post.get_title()]
-        text_bits.append(strip_tags(abstract))
-        text_bits.append(post.get_description())
-        text_bits.append(' '.join(post.get_keywords()))
+        if description:
+            text_bits.append(description)
+        if keywords:
+            text_bits.append(' '.join(keywords))
         for category in post.categories.all():
             text_bits.append(
                 force_text(category.safe_translation_getter('name')))
         for tag in post.tags.all():
             text_bits.append(force_text(tag.name))
-        if post.content:
+
+        if get_setting('USE_PLACEHOLDER'):
             plugins = post.content.cmsplugin_set.filter(language=language)
+            content_bits = []
             for base_plugin in plugins:
                 content = get_plugin_index_data(base_plugin, request)
-                text_bits.append(' '.join(content))
-        for attribute in optional_attributes:
-            value = force_text(getattr(post, attribute))
-            if value and value not in text_bits:
-                text_bits.append(value)
-        return ' '.join(text_bits)
+                content_bits.append(' '.join(content))
+            post_text = ' '.join(content_bits)
+        else:
+            post_text = post.safe_translation_getter('post_text')
+            if post_text:
+                post_text = strip_tags(post_text)
+        self.prepared_data['post_text'] = post_text
+        text_bits.append(post_text)
 
-    def prepare_fields(self, post, language, request):
-        super(PostIndex, self).prepare_fields(post, language, request)
-        data = [self.prepared_data['text']]
-        self.prepared_data['keywords'] = ' '.join(post.get_keywords())
-        self.prepared_data['tags'] = ' '.join(post.get_tags())
-        self.prepared_data['post_text'] = ' '.join(post.safe_translation_getter('post_text'))
-        self.prepared_data['text'] = ' '.join(data)
```

Comment:
```tags``` are filled via ```model_attr``` while ```post_text``` and ```text``` are handled in ```get_search_data```. Seems I messed up with ```tags```. To be fixed in the next commit.